new-frontiers-14/frontier-station-14
# POI Spawning: Handle empty preset list, restrict protos in NfAdventureRule

## About the PR



* Better handles empty preset lists in NfAdventureRuleSystem and PointOfInterestSystem
* ~~Removes the preset lists from most POIs (note: not the cove).~~ Reverted this.
  * ~~If this should be returned to how it was ("we don't want to support secret/deathmatch/X"), easily done, no harm no foul.~~ Reverted this.  No harm, no foul.
* Adds a comment to the PointOfInterestPrototype that empty preset lists are treated as accepting any preset
* Reorganized namespace/system dependencies in alphabetical order
  * ~~milon~~ Reverted this.

## Why / Balance


Empty preset lists should be supported in-spec, and it makes sense (at least to me) to have it as working in all presets if the NfAdventureRule is present.

If this was the case, #2613 wouldn't have been needed.

## How to test


Run the game, check that all of the POI stations are present, and the cove is present if running NFPirates.

## Media


Behold, stations.
![image](https://github.com/user-attachments/assets/eda2cc38-5fe3-40bc-9f1c-3982123ca7ee)

## Requirements



## Breaking changes


**Changelog**


No player-facing changes, no changelog.
